release/downloadIstioCandidate.sh

```diff
@@ -87,6 +92,10 @@ with_arch() {
 
 without_arch() {
   printf "\nDownloading %s from %s ..." "$NAME" "$URL"
+  if ! curl -o /dev/null -sIf "$URL"; then
+    printf "\n%s is not found, please specify a valid ISTIO_VERSION and TARGET_ARCH\n" "$URL"
+    exit
```

Comment:
as this is without arch, we might need to replace `TARGET_ARCH` with `OS`?  or just print "xxx is not found, please specify a valid ISTIO_VERSION"